math/mathcore/inc/Math/Functor.h

```diff
@@ -158,11 +158,31 @@ class GradFunctor : public IGradientFunctionMultiDim  {
    {}
 
    /// Construct from a pointer to member function and member function types for function and derivative evaluations.
-   template <class PtrObj, typename MemFn, typename GradMemFn>
-   GradFunctor(const PtrObj& p, MemFn memFn, GradMemFn gradFn, unsigned int dim )
-      : fDim{dim}, fFunc{std::bind(memFn, p, std::placeholders::_1)}, fDerivFunc{std::bind(gradFn, p, std::placeholders::_1, std::placeholders::_2)}
+   template <class PtrObj, typename MemFn, typename DerivMemFn,
+             std::enable_if_t<std::is_floating_point<decltype((std::declval<std::remove_pointer_t<PtrObj>>().*
```

Comment:
It's great that you found a way to do it with `enable_if`! I also couldn't figure it out how to do it less verbose. Maybe let's leave it like you have it now, and then revisit it when C++17 becomes the minimum standard.